### PR TITLE
PWA

### PR DESCRIPTION
**Pull request by anthonyj33**
# PWA Features
- Made into installable PWA
- Push notifications work w/ API
    - If notifications supported (and not granted), prompts on click
    - Once user grants permission, subscribes and passes subscription to API w/ group `announcements`
    - When push is received, shows notification
- Stack (PR #2)

### CODE REVIEW

**Comment by anthonyj33** [comment]
@Awesome-E please review + verify that all API calls are correct

*Should only call subscribe endpoint once now* -- you can verify this as well

**Comment by Awesome-E** [comment]
- PWA calling `/subscribe` only once is working
- Notifications work
- Reviewing code now

File: src/main.js
Comment (Awesome-E) [inline src/main.js]: Same as comment below; guard clauses help with not needing so many nested scopes

File: src/main.js
Comment (Awesome-E) [inline src/main.js]: Since this is inside of a function where you only care for `permission !== 'granted'`, it's better to use a guard clause

`if (permission === 'granted') return`
Then do the request logic with a layer less of nesting/indentation

**Review by Awesome-E — APPROVED** [review]
Fix these two format/style things then feel free to merge. Usually I'd request changes, but this is small enough that I won't be too mad otherwise, plus less friction knowing you can just fix and not have to wait for another review

**Comment by Awesome-E** [comment]
Just fix the two things above first. Approved so that you don't have to wait for another review to happen.

If you'd still like me to quick look that to catch a stupid mistake, then re-request review

**Comment by anthonyj33** [comment]
Thanks for pointing that out.. will fix before merging

I always neglect guard clauses..

**Pull request merged**
